breathecode/events/permissions/consumers.py

```diff
@@ -61,7 +62,8 @@ def event_by_url_param(context: PermissionContextType, args: tuple, kwargs: dict
         | Q(cohort__available_as_saas=None, cohort__academy__available_as_saas=False),
         user=request.user).exists()
 
-    if not is_host and (not is_free_for_bootcamps or not user_with_available_as_saas_false):
+    if not is_host and not is_free_for_all and (not is_free_for_bootcamps
```

Comment:
Add validation in the model, if it's free for all it's free for bootcamp too, set in these cases `free_for_bootcamps` to `None`, then apply an or condition `(event.free_for_all or event.free_for_bootcamps)`